hummingbot/wallet/ethereum/web3_wallet_backend.py

```diff
@@ -296,11 +303,12 @@ def stop(self):
             self._check_transaction_receipts_task = None
 
     async def check_network(self) -> NetworkStatus:
+        # Assume connected if received new blocks in last 2 minutes
+        if time.time() - self._last_timestamp_received_blocks > 60 * 2:
```

Comment:
This is wrong... while block times average 15 seconds per block, it's not rare for some blocks time to be more than 2 minutes. Block times are expected to form a Gaussian random distribution, rather than being perfectly regular.

You should use the last successful block check's timestamp instead, not the new block's timestamp.

Reply:
this is just a quick check, if we received a block within the last 2 min, assume our connection is still good, if not ping for gas price just to verify that connection to web3 provider is still good.

Reply:
ok, I think I get it then. It wasn't clear when looking at just the diffs.